dataprofiler/data_readers/csv_data.py

```diff
@@ -89,39 +97,40 @@ def __init__(self, input_file_path=None, data=None, options=None):
                 self._quotechar = self._default_quotechar
 
     @property
-    def selected_columns(self):
+    def selected_columns(self) -> List[str]:
         """Return selected columns."""
         return self._selected_columns
 
     @property
-    def delimiter(self):
+    def delimiter(self) -> Optional[str]:
         """Return delimiter."""
         return self._delimiter
 
     @property
-    def quotechar(self):
+    def quotechar(self) -> Optional[str]:
         """Return quotechar."""
         return self._quotechar
 
     @property
-    def header(self):
+    def header(self) -> Optional[Union[str, int]]:
         """Return header."""
         return self._header
 
     @property
-    def is_structured(self):
+    def is_structured(self) -> bool:
         """Determine compatibility with StructuredProfiler."""
         return self.data_format == "dataframe"
 
-    def _check_and_return_options(self, options):
+    @staticmethod
+    def _check_and_return_options(options: Optional[Dict]) -> Dict:
         """
         Ensure options are valid inputs to the data reader.
 
         :param options: dictionary of options for the csv reader to validate
         :type options: dict
         :return: None
         """
-        options = super()._check_and_return_options(options)
+        options = super(CSVData, CSVData)._check_and_return_options(options)
```

Comment:
What's going on here? this seems abnormal.

Reply:
I had to change `_check_and_return_options` to be a static method because `BaseData` implements it as a static method. However, `super()` with no arguments only really works with instance and class methods. I ended up following https://stackoverflow.com/questions/26788214/super-and-staticmethod-interaction#:~:text=When%20you%20call%20a%20class,or%20class%20it%20was%20called. to figure out how to make it work with static methods.

Reply:
Instead could we use `BaseData. _check_and_return_options(options)`?

Reply:
@JGSweets thoughts on this? Any additional thoughts around this implementation? 

Found this example which helps provide an example for how @Sanketh7 implement this: https://stackoverflow.com/questions/26788214/super-and-staticmethod-interaction#:~:text=class%20Second(First)%3A%0A%20%20%40staticmethod%0A%20%20def%20getlist()%3A%0A%20%20%20%20l%20%3D%20super(Second%2C%20Second).getlist()%20%20%23%20note%20the%202nd%20argument%0A%20%20%20%20l.append(%27second%27)

Reply:
I'm fine with it. We can refactor in the future if necessary.